runtime/include/chpl-comm-compiler-macros.h

```diff
@@ -61,6 +62,28 @@ void chpl_gen_comm_get(void *addr, c_nodeid_t node, void* raddr,
     chpl_comm_get(addr, node, raddr, size, commID, ln, fn);
   }
 }
+#ifdef HAS_GPU_LOCALE
+static inline
+void chpl_gen_comm_get_gpu(void *addr, c_nodeid_t src_node,
+                           c_sublocid_t src_subloc, void* raddr, size_t size,
+                           int32_t commID, int ln, int32_t fn)
+{
+  c_sublocid_t dst_subloc = chpl_task_getRequestedSubloc();
+
+  if (chpl_nodeID == src_node) {
+    chpl_gpu_memcpy(dst_subloc, addr, src_subloc, raddr, size, commID, ln, fn);
```

Comment:
Yeah, I am a bit more reluctant to a rename here.

Reply:
Do you want to argue for a change?

Reply:
Well it's one of those thing I'm not particularly happy with but I don't have any constructive suggestions so I'm willing to shrug it off after complaining a bit (I dislike my suggestion of `chpl_gpu_aware_memcpy` too since it would be mixing the "name" of the function with the convention of having `chpl_gpu` serve as our "poor man's module namespacing" identifier (and `chpl_gpu_gpu_aware_memcpy` is just.. well. no.).

But to voice my complaint (so I can put it rest): if I was new to the project and you told me there was two functions: `chpl_gpu_memcpy` and `chpl_memcopy` and asked me to guess what these two do.  I would guess that `chpl_memcopy` would be the more "abstract" version that would figure out what kind of copy you need to do and dispatch to, and `chpl_gpu_memcpy` would be that more "concrete" function that gets called if/when a GPU is involved (rather than the other way around).

Reply:
> But to voice my complaint (so I can put it rest): if I was new to the project and you told me there was two functions: chpl_gpu_memcpy and chpl_memcopy and asked me to guess what these two do. I would guess that chpl_memcopy would be the more "abstract" version that would figure out what kind of copy you need to do and dispatch to, and chpl_gpu_memcpy would be that more "concrete" function that gets called if/when a GPU is involved (rather than the other way around).

Totally reasonable.

Some alternatives as food for thought:

- `chpl_gpu_comm`
- `chpl_gpu_comm_get`/`chpl_gpu_comm_put`

(though I imagine we'll probably need something more `comm`-y down the road)

- `chpl_gpu_data_copy` 

(tries too hard to avoid just calling it `memcpy` imo)

- `chpl_memcpy_subloc`

(is interesting, but doesn't have `chpl_gpu` prefix. `chpl_gpu_memcpy_subloc` is too tautological)

I am still going to stick with what I have unless you want to champion either one of these or something else.

Reply:
I guess in the future, we may have `chpl_memcpy_subloc` elsewhere in the runtime. That may end up calling some `chpl_gpu` function..

Reply:
> I am still going to stick with what I have unless you want to champion either one of these or something else.

Nope, I'm fine, just keep these ideas in the back-of-your-head incase we have another opportunity to rename/refactor it.